text/3371-cargo-target-dir-templates.md

```diff
@@ -0,0 +1,423 @@
+- `"auto"`: create the symlink, produce a warning (or note) but do not fail the command
+- `false`: don't create the symlink at all (don't touch it if it exists already though)
+
+### Advantages of forward links
+
+This has a two big advantages: not breaking external tools and giving easy access to artifacts produced by `cargo build/test/doc` to users (they're in the habit of typing `./target/debug/my-bin`, this would continue working with forward links).
+
+### Drawbacks of forward links
+
+There a few cases where a symlink instead of a real dir will break programs: at least SQLite 3 can be configured to raise an error if the database is behind a symlink anywhere in its opening path, it's probably other programs can also be configured to check this (or do it by default). Since templated `CARGO_TARGET_DIR` won't become a default in this RFC, we are not breaking any existing use cases.
+
+Additionally, there are issue when competing `CARGO_TARGET_DIR`s are set, for example if a different one is used for rust-analyzer and for nightly/stable versions:
+
+- Do we update the symlink all the time ? Rust-Analyzer and a `cargo build` could be running in parallel, with unpredictable results
+- Do we only set it if it doesn't exist yet ? It could become outdated, a tool could run first and make it point somewhere unexpected
+
+Another drawback is about the warning/note when creating the symlink fails:
+
+- Do we warn/notify on each command ? Only on creation ? Only with `--verbose` ? All have issues with either being too verbose or users potentially missing relevant information. This can be determined during stabilization though, once more people have experience with it, and is easy to change later on.
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+<!--
+Think about what the natural extension and evolution of your proposal would
+be and how it would affect the language and project as a whole in a holistic
+way. Try to use this section as a tool to more fully consider all possible
+interactions with the project and language in your proposal.
+Also consider how this all fits into the roadmap for the project
+and of the relevant sub-team.
+
+This is also a good place to "dump ideas", if they are out of scope for the
+RFC you are writing but otherwise related.
+
+If you have tried and cannot think of any future possibilities,
+you may simply state that you cannot think of anything.
+
+Note that having something written down in the future-possibilities section
+is not a reason to accept the current or a future RFC; such notes should be
+in the section on motivation or rationale in this or subsequent RFCs.
+The section merely provides additional information.
+-->
+
+- Introduce remapping into the concept in some way.
+- Introduce a form of garbage collection. Expanded upon this [Zulip thread](https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/last-use.20tracking).
```

Comment:
```suggestion
- Introduce a form of garbage collection for these target directories so we don't leak them when projects are deleted, see rust-lang/cargo#13136
```
- Updating of link
- Clarifying why this is relevant

Reply:
Done in https://github.com/rust-lang/rfcs/pull/3371/commits/2ae536e3d73ae211b0342aa0448f95123a240516